### ISSUE TITLE
about the training details

### ISSUE THREAD

**Issue by LifangD** [tianran/glimvec#1]
Where is the training details ?  Could you please  explain the structure and steps of the files?
Thanks~

**Comment by tianran** [comment]
Hi~ Sorry I didn't take much time to write the documentation (which I should do and will, though...) So can you tell me more exactly what do you want to know? Generally, if you want to train a model, simply run trainKB or trainKB.py with the -h option. Then it will display a help message about how to use the command. :) Usually the default setting works well, so you might only want to change the --para option, which specifies the number of parallel threads for training, and the --numBatches option, which specifies how long the training process will go on. I'm always here to answer your questions :)

**Comment by LifangD** [comment]
Hi, I read the paper about joint training on KBC task, which gives me a lot of inspiration. And I noticed that some details, such as how to combine the two parts's loss and how to design the learning rate, are of great importance. So I am very interested in learning these tricks. And besides, there are some intermediate results which I can't directly link to the source code. 
![image](https://user-images.githubusercontent.com/33337947/49263360-3150b280-f485-11e8-88f3-82d4d397967f.png)
It would be very helpful if you could specify the steps of generating these results and the related files.
Thanks~

**Comment by tianran** [comment]
Thanks for your interest :) All right, do you mean you want to understand the source code, the structure of model files, or want to know how to generate this specific model-base model? If you want to understand the source code, especially the part related to training tricks, we have to look at cpp/TainerKB.cpp and cpp/HyperParametersKB.h they are not very easy to understand; and some code no longer looks good for me now, because I’m also learning and experimenting. But anyway, I can explain the code if you like. If you want to know about the model files, generally the .npy files are model parameters stored in the format of numpy arrays that can be directly loaded to python. Those named with *steps.npy are training steps used in SGD. As we mentioned in the paper, different components of our model use different step counters. If you want to know how to generate this specific model-base model, we have to change the source code a little bit because the current source code is for model-joint. It’s no big deal, though, all we have to do is to change a flag in cpp/HyperParametersKB.h and recompile the source. Have you been able to successfully compile the source code?

**Comment by LifangD** [comment]
Thank you so much, I'll try~
